### PR TITLE
CDAP-6493 Add a max limit for reading log events from Kafka

### PR DESCRIPTION
**Pull request by poornachandra**
JIRA - https://issues.cask.co/browse/CDAP-6493
Build - http://builds.cask.co/browse/CDAP-DUT4404-1

### CODE REVIEW

File: cdap-watchdog/src/main/java/co/cask/cdap/logging/read/KafkaLogReader.java [hunk 1]
```diff
@@ -44,6 +44,8 @@
 public class KafkaLogReader implements LogReader {
   private static final Logger LOG = LoggerFactory.getLogger(KafkaLogReader.class);
   private static final int KAFKA_FETCH_TIMEOUT_MS = 30000;
+  // Maximum events to read from Kafka for any call
+  private static final int MAX_READ_EVENTS_KAFKA = 10000;
```
Comment (albertshau) [inline cdap-watchdog/src/main/java/co/cask/cdap/logging/read/KafkaLogReader.java:48]: do you see any need to make this a configuration setting?

**Comment by albertshau** [comment]
one question, otherwise lgtm

File: cdap-watchdog/src/main/java/co/cask/cdap/logging/read/KafkaLogReader.java (hunk 1 above)
Comment (poornachandra) [inline cdap-watchdog/src/main/java/co/cask/cdap/logging/read/KafkaLogReader.java:48]: Initially I thought about making it configurable, but then I realized that if we are not able to find even one log event after scanning 10k records then increasing it to 20k may not make much difference. 

What do you say? I don't have a strong opinion either way.

Comment (albertshau) [inline cdap-watchdog/src/main/java/co/cask/cdap/logging/read/KafkaLogReader.java:48]: as we discussed, if this limit is being hit, changing the number of log partitions would fix the underlying issue, so we will not have a configuration setting for this.

**Comment by albertshau** [comment]
lgtm

**Pull request merged**